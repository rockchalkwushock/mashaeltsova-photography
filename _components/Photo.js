import LazyLoad from 'react-lazyload'

import { createUrl } from '../lib'

const Photo = ({ id }) => (
  <LazyLoad height={200} once>
    <picture>
      <source media="(min-width: 1024px)" srcSet={createUrl(false, id, 320)} />
      <source media="(min-width: 768px)" srcSet={createUrl(false, id, 360)} />
      <source
        media="(min-width: 600px) and (orientation: landscape)"
        srcSet={createUrl(false, id, 275)}
      />
      <source media="(min-width: 540px)" srcSet={createUrl(false, id, 475)} />
      <source media="(min-width: 425px)" srcSet={createUrl(false, id, 375)} />
      <img src={createUrl(false, id, 300)} />
      <style jsx>{`
        img {
          -webkit-box-shadow: 6px 6px 6px rgba(54, 69, 79, 0.7);
          box-shadow: 6px 6px 6px rgba(54, 69, 79, 0.7);
          border-radius: 15px;
        }
      `}</style>
    </picture>
  </LazyLoad>
)

export default Photo
