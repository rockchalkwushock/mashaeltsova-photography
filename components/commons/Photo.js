import { Image, Transformation } from 'cloudinary-react'
import PropTypes from 'prop-types'

/**
 * REVIEW
 *
 * family - 8
 * portrait - 10
 * travel - 10
 * wedding - 8
 *
 */

const Photo = ({ publicId }) => (
  <div>
    <Image publicId={publicId} secure use_root_path>
      <Transformation
        crop="scale"
        dpr="auto"
        fetch_format="auto"
        quality="auto:best"
        width="320"
      />
    </Image>
    <style jsx>{`
      div {
        grid-column: span 8;
      }
      @media (min-width: 768px) {
        div {
          grid-column: span 12;
        }
      }
      @media (min-width: 1024px) {
        div {
          grid-column: span 12;
        }
      }
    `}</style>
  </div>
)

Photo.propTypes = {
  publicId: PropTypes.string.isRequired
}

export default Photo
