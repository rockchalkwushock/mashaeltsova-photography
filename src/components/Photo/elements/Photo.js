import React from 'react'
import Img from 'gatsby-image'
import styled from 'styled-components'

const UnstyledPhoto = props => <Img {...props} />

const Photo = styled(UnstyledPhoto)`
  border: none;
  box-shadow: 2px 2px 10px 1px rgba(0, 0, 0, 1);
  height: 240px;
  width: 240px;

  @media (min-width: ${({ theme }) => theme.sizes.sm}) {
    width: 400px;
  }
  @media (min-width: ${({ theme }) => theme.sizes.md}) {
    width: 300px;
  }
`
Photo.displayName = 'Photo'

export default Photo
