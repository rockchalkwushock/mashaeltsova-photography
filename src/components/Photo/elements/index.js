import React from 'react'
import Img from 'gatsby-image'
import styled from 'styled-components'

const UnstyledPhoto = props => <Img {...props} />

const StyledPhoto = styled(UnstyledPhoto)`
  border: none;
  box-shadow: 2px 2px 10px 1px rgba(0, 0, 0, 1);
`

StyledPhoto.displayName = 'StyledPhoto'

export default StyledPhoto
