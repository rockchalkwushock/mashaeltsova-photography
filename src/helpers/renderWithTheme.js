import React from 'react'
import { mount } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import theme from '../utils/theme'

/**
 * @fileOverview
 *
 * HOC for testing components with `theme` prop being used.
 */

const renderWithTheme = Component =>
  mount(<ThemeProvider theme={theme}>{Component}</ThemeProvider>)

export default renderWithTheme
