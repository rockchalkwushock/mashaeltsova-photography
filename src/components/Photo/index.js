import React from 'react'

import { Photo } from './elements'

const AppPhoto = ({ alt, photo, title }) => (
  <Photo alt={alt} sizes={photo.sizes} title={title} />
)

export default AppPhoto
