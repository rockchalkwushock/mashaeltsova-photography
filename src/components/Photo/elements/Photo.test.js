import React from 'react'

import { renderWithTheme } from '../../../helpers'
import Photo from './Photo'

describe('Styled Component: <Photo />', () => {
  let wrapper
  beforeAll(() => {
    wrapper = renderWithTheme(<Photo sizes={{}} />)
  })
  test('should render markup', () => {
    expect(wrapper).toMatchSnapshot()
  })
  // REVIEW
  // Cannot see styles???
  test.skip('should have styles', () => {
    expect(wrapper).toHaveStyleRule('border', 'none')
    expect(wrapper).toHaveStyleRule(
      'box-shadow',
      '2px 2px 10px 1px rgba(0,0,0,1)'
    )
    expect(wrapper).toHaveStyleRule('height', '240px')
    expect(wrapper).toHaveStyleRule('width', '240px')
    expect(wrapper).toHaveStyleRule('width', '400px', {
      media: '(min-width:540px)'
    })
    expect(wrapper).toHaveStyleRule('width', '300px', {
      media: '(min-width:736px)'
    })
  })
})
