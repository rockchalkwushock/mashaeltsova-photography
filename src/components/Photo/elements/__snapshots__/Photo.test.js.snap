// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styled Component: <Photo /> should render markup 1`] = `
.c0 {
  border: none;
  box-shadow: 2px 2px 10px 1px rgba(0,0,0,1);
  height: 240px;
  width: 240px;
}

@media (min-width:540px) {
  .c0 {
    width: 400px;
  }
}

@media (min-width:736px) {
  .c0 {
    width: 300px;
  }
}

<ThemeProvider
  theme={
    Object {
      "sizes": Object {
        "lg": "1200px",
        "md": "736px",
        "sm": "540px",
      },
    }
  }
>
  <Photo
    sizes={Object {}}
  >
    <UnstyledPhoto
      className="c0"
      sizes={Object {}}
    >
      <Image
        Tag="div"
        alt=""
        className="c0"
        fadeIn={true}
        sizes={Object {}}
      >
        <div
          className=" gatsby-image-outer-wrapper"
          style={
            Object {
              "position": "relative",
            }
          }
        >
          <div
            className="c0 gatsby-image-wrapper"
            style={
              Object {
                "overflow": "hidden",
                "position": "relative",
              }
            }
          >
            <div
              style={
                Object {
                  "paddingBottom": "NaN%",
                  "width": "100%",
                }
              }
            />
            <Img
              alt=""
              onLoad={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <img
                alt=""
                onLoad={[Function]}
                style={
                  Object {
                    "height": "100%",
                    "left": 0,
                    "objectFit": "cover",
                    "objectPosition": "center",
                    "opacity": 1,
                    "position": "absolute",
                    "top": 0,
                    "transition": "opacity 0.5s",
                    "width": "100%",
                  }
                }
              />
            </Img>
            <noscript
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<img width=\\"\\" height=\\"\\" src=\\"undefined\\" srcset=\\"undefined\\" alt=\\"\\" title=\\"\\" sizes=\\"\\" style=\\"position:absolute;top:0;left:0;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/>",
                }
              }
            />
          </div>
        </div>
      </Image>
    </UnstyledPhoto>
  </Photo>
</ThemeProvider>
`;
